### ISSUE TITLE
MRL-0803: authorize and verify Experiment-0 contamination and held-out isolation evidence

### ISSUE THREAD

**Issue by TheHalfMoon** [TheHalfMoon/MESC#407]
## Purpose

Advance the canonical Rule-R4 MRL frontier from closed model/corpus identity gates into genuine contamination and held-out isolation evidence for the current Experiment-0 foundation tournament.

This unit begins only after Issue #404 closed `completed` with fresh post-merge qualification.

```text
CANONICAL_MAIN_SHA = cfb3fa6f962da7605a3d7268b029760869436918
CANONICAL_MAIN_TREE = e7f7375df5901f3a3a5484786a7b7166387682ed
PREDECESSOR_ISSUE = #404 / CLOSED_COMPLETED
MRL-0800 = CLOSED_CANONICAL
MRL-0801 = CLOSED_CANONICAL
MRL-0802 = CLOSED_CANONICAL
MRL-0699 = CLOSED_CANONICAL
MRL-0803 = PLANNED
```

The exact admitted synthetic corpus identity from canonical MRL-0802 is:

```text
CORPUS_SHA256 = 977f5faa543f479276ad3742af797435e2ee732d8b9e8e594fabbf99e916b16e
CORPUS_BYTE_COUNT = 3281
RECORD_COUNT = 16
MRL_0802_EVIDENCE_SHA256 = 1d6d14590a19c20bcd794e4c0ddbd2fa5e1c767b70e9d199fc169aeaaa86b762
RIGHTS_EVIDENCE_SHA256 = cdec260d0a4f08741905875a7fd3af220fd246cd4648c2b21cc14f789f72b88f
PROVENANCE_SHA256 = e38a77173117d78aa3c24d0a73ce771b7f6b954a0293b12b27d09667995d7884
```

## Scientific boundary

Experiment-0 is a no-training foundation evaluation. MRL-0803 must prove contamination/isolation only for MESC-controlled corpus, training, search, prompt-development, replication, and sealed-evaluation surfaces. It MUST NOT claim knowledge of, or absence of overlap with, inaccessible upstream pretraining corpora for Qwen or Gemma.

A valid PASS must prove that the sealed evaluation surface is excluded from every MESC-controlled training/adaptive-search surface and that the evaluation partition is reproducibly bound to the exact admitted corpus.

## Authorized implementation scope

Within this single Rule-R4 issue, execute dependency-ordered work through separate guarded PRs as needed:

1. add a deterministic MRL-0803 isolation authorization artifact bound to the exact canonical corpus and Experiment-0;
2. add a deterministic producer/verifier that consumes the exact admitted corpus bytes and emits path-free isolation evidence;
3. freeze a deterministic patient-level split with synthetic-household grouping so no patients sharing the same normalized synthetic household/address can cross evaluation tiers;
4. produce explicit Tier 1 search, Tier 2 replication, and Tier 3 sealed identities, with Tier 3 item-level access prohibited to the adaptive research process;
5. prove exact-record and household disjointness across tiers and record a deterministic near-structural contamination analysis under frozen thresholds;
6. bind corpus identity, split/held-out identity, lineage report identity, decontamination report identity, and sealed-evaluation exclusion into the existing `mesc.mrl.real_preflight.isolation.v1` evidence contract;
7. preserve uncertainty/fail-closed behavior if any required semantic contamination detector is not scientifically justified by the available structural corpus;
8. independently revalidate genuine produced bytes before any trust mutation;
9. only after genuine evidence is independently verified, admit the exact evidence digest through the smallest separate trust/index/checklist mutation;
10. require fresh post-merge qualification proving `MRL-0803 = CLOSED_CANONICAL` before this issue closes.

The initial producer/authorization PR MUST NOT mutate `TRUSTED_MRL_REAL_PREFLIGHT_EVIDENCE_SHA256`, the canonical real-preflight evidence index, or the MRL-0803 checklist state.

## Isolation policy requirements

At minimum:

- every input record must be a canonical Patient projection from the exact admitted corpus;
- patient IDs must be unique;
- exact duplicate records across tiers are prohibited;
- normalized synthetic household/address groups are indivisible across tiers;
- tier membership must be deterministic from canonical content identity and a committed policy identity, never filesystem order or runtime randomness;
- the target partition must be exact and bounded for the 16-record corpus;
- Tier 3 membership/content must not become adaptive model-development feedback;
- Tier 3 must be excluded from MESC-controlled training and search surfaces;
- any future training lineage must remain subject to canonical MRL-0601..0607 exact/near/semantic contamination controls and no-canary-recycling rules;
- no empty or fabricated training lineage may be created merely to satisfy an interface.

## Genuine evidence contract

A final evidence candidate must satisfy the current canonical MRL-0803 payload:

```text
kind = mesc.mrl.real_preflight.isolation.v1
contamination_disposition = PASS
corpus_sha256 = 977f5faa543f479276ad3742af797435e2ee732d8b9e8e594fabbf99e916b16e
decontamination_report_sha256 = exact produced report
heldout_evaluation_sha256 = exact frozen held-out/sealed identity
lineage_report_sha256 = exact produced lineage/isolation report
sealed_evaluation_excluded_from_training = true
```

Schema-valid evidence is not trusted evidence by itself. Production trust admission requires independent verification and a separate exact-digest repository mutation.

## Qualification

Before each guarded merge require the applicable subset of:

- exact corpus/hash/record-count binding;
- deterministic split identity and repeated-run byte identity;
- exact duplicate and household-crossing rejection tests;
- malformed/non-Patient/corpus-drift rejection;
- Tier-3 search/training exposure rejection;
- fail-closed contamination disposition tests;
- branch-local machine state remains noncanonical until trust admission;
- Ruff lint/format;
- strict mypy;
- docs hygiene;
- full pytest Python 3.11/3.12;
- MRL machine-state drift/manual-edit gate;
- `medscale check`;
- exact-head CodeQL;
- fresh independent substantive exact-head review;
- zero unresolved material review findings/threads;
- active ruleset verification;
- guarded expected-head merge using merge-commit semantics only;
- fresh post-merge qualification before evidence execution, trust admission, and final closure as applicable.

## Explicit non-authority

This issue does not itself authorize model/tokenizer loading, inference, GPU execution, Experiment-0 execution, training, fine-tuning, weight mutation, model promotion, release, or clinical deployment.

```text
MRL_REAL_EXPERIMENT_READY = FALSE
MODEL_LOADING_AUTHORIZED = FALSE
TOKENIZER_LOADING_AUTHORIZED = FALSE
INFERENCE_AUTHORIZED = FALSE
GPU_EXECUTION_AUTHORIZED = FALSE
EXPERIMENT_0_EXECUTION_AUTHORIZED = FALSE
TRAINING_AUTHORIZED = FALSE
TRAINING_READY = FALSE
PROJECT_COMPLETION = NOT CLAIMED
```

Do not close this issue until the exact MRL-0803 evidence digest is canonically trusted and fresh post-merge machine-state qualification proves `MRL-0803 = CLOSED_CANONICAL`.

**Comment by TheHalfMoon** [comment]
MRL-0803 producer implementation is now proposed in PR #408 at exact head `30eaf216c6f4b809b59dc659fad19011eb4cbd4b` (tree `d81ccab8ff3d81a23b649f0b1e0f32adf7ea9f91`).

Local qualification before push:

```text
Ruff = PASS
Ruff format = PASS
strict mypy = PASS / 455 source files
docs hygiene = PASS / 122 Markdown files
focused producer tests = 34 passed
broader MRL regression suite = 112 passed
medscale check = CLEAN
git diff --check = PASS
production trust/index/checklist mutation sentinel = PASS
```

A development-only run against the exact admitted corpus confirmed deterministic 8/4/4 partitioning with zero exact cross-tier duplicates and zero synthetic-household crossings. Its evidence digest is intentionally NON-ADMISSIBLE because the producer has not yet been canonically merged. Genuine evidence will be regenerated only after guarded merge and fresh post-merge qualification of the producer.

No model/tokenizer loading, inference, GPU execution, Experiment-0 execution, training, trust admission, or project-completion authority is granted by PR #408.

**Comment by TheHalfMoon** [comment]
Governance correction: Issue #407 was automatically closed at the producer PR #408 merge, before the issue's explicit closure conditions were satisfied. That closure was non-canonical and has been reverted.

Current state remains:
- producer PR #408 merged to canonical main `d584b4e8e80f953b461b0a09ed7dfb00f10888ab`;
- genuine post-merge MRL-0803 evidence has not yet been produced/admitted;
- the exact MRL-0803 evidence digest is not yet in the production trust registry/index;
- MRL-0803 checklist state remains open;
- fresh post-trust-admission machine-state qualification has not yet proved `MRL-0803 = CLOSED_CANONICAL`.

Rule R4 therefore keeps #407 as the sole active MRL frontier. No MRL-0804 successor work is authorized until #407 satisfies its stated closure rule.

**Comment by TheHalfMoon** [comment]
MRL-0803 trust-admission progress record — no closure claim

The genuine post-producer MRL-0803 evidence has been produced from canonical producer commit `d584b4e8e80f953b461b0a09ed7dfb00f10888ab` and independently revalidated from a separate clean exact-source worktree.

```text
CORPUS_SHA256 = 977f5faa543f479276ad3742af797435e2ee732d8b9e8e594fabbf99e916b16e
MRL_0803_EVIDENCE_SHA256 = 629d8d39a6ea44e753b6d83e07116b12213a74c14f395bea700224fc6781c8e2
SPLIT_MANIFEST_SHA256 = 841577a1d62c1e79464465246b73f19e6b2908b0c170990351cc86b4ae205f63
LINEAGE_REPORT_SHA256 = 6137bde514eec879482c62cd2e0920be6095d7623ec5bf886b66b9523b0e043d
DECONTAMINATION_REPORT_SHA256 = 26baf19d18e3af4cab3eb61df2a65098e0829bcde3c484df530dfdc590293647
HELDOUT_EVALUATION_SHA256 = a74ca84315459622b38a437fd31b9b82058701c5a28798ced3c8deac5a5f9247
TIER_COUNTS = 8,4,4
CONTAMINATION_DISPOSITION = PASS
SEALED_EVALUATION_EXCLUDED_FROM_TRAINING = true
INDEPENDENT_EVIDENCE_RELATION_CHECK = PASS
```

Separate trust admission PR #409 is open at exact head:

```text
HEAD_SHA = ab435fe970ccd4db7593b6721b8d3c34abd2fdb2
HEAD_TREE = dce64b8c75cb10773d21388b7c1cc83be4434b79
CHANGED_FILES = 5
```

Exact-head local qualification on that committed head, with runtime imports pinned to the same worktree, is complete:

```text
Ruff = PASS
Ruff format = PASS / 948 files
strict mypy = PASS / 455 source files
docs hygiene = PASS / 122 Markdown files
trust + machine-state + MRL-0803 targeted tests = 64 passed
MRL machine-state CI gate = 14 passed
medscale check = CLEAN
git diff --check = PASS
worktree cleanliness = PASS
```

Fresh independent substantive review of exact head `ab435fe970ccd4db7593b6721b8d3c34abd2fdb2` reported zero material findings. Review thread count is zero. Exact-head CodeQL is PASS.

Hosted CI run `34706224129` is still on its original attempt. `quality (py3.11)` and `quality (py3.12)` are currently executing full `Pytest`; no rerun, cancellation, or rerun-to-green has been performed.

Therefore:

```text
PR_409_MERGE = NOT YET AUTHORIZED
MRL_0803 = NOT YET CLOSED_CANONICAL
ISSUE_407 = MUST REMAIN OPEN
MRL_0804 = BLOCKED_BY_RULE_R4
MODEL_LOADING_AUTHORIZED = FALSE
INFERENCE_AUTHORIZED = FALSE
GPU_EXECUTION_AUTHORIZED = FALSE
EXPERIMENT_0_EXECUTION_AUTHORIZED = FALSE
TRAINING_AUTHORIZED = FALSE
TRAINING_READY = FALSE
PROJECT_COMPLETION = NOT CLAIMED
```

The next permitted action is guarded merge-commit of PR #409 only if the original exact-head required hosted checks complete successfully, followed by fresh post-merge canonical machine-state qualification proving `MRL-0803 = CLOSED_CANONICAL` before this issue is closed.

**Comment by TheHalfMoon** [comment]
## Post-merge closeout correction

PR #409 merged with guarded expected-head merge semantics at merge commit `daa2e83774b7bc210014e4e160bcf1203e943a4e` (tree `dce64b8c75cb10773d21388b7c1cc83be4434b79`; parents `d584b4e8e80f953b461b0a09ed7dfb00f10888ab` and `ab435fe970ccd4db7593b6721b8d3c34abd2fdb2`).

GitHub automatically closed this issue at the merge boundary before fresh post-merge qualification had completed. This issue explicitly permits `completed` closure only after fresh post-merge canonical qualification proves `MRL-0803 = CLOSED_CANONICAL`, so the issue has been reopened fail-closed.

Fresh post-merge workflows on exact merge SHA `daa2e83774b7bc210014e4e160bcf1203e943a4e` are now the active closeout evidence:

- CI run `34709912361` — post-merge push / in progress at this checkpoint
- CodeQL run `34709912267` — post-merge push / in progress at this checkpoint
- Optional Extras / Backends run `34709912368` — post-merge push / in progress at this checkpoint

The exact trusted MRL-0803 evidence digest remains:

```text
629d8d39a6ea44e753b6d83e07116b12213a74c14f395bea700224fc6781c8e2
```

No post-merge success, `CLOSED_CANONICAL` state, MRL-0804 authority, model loading, inference, GPU execution, Experiment-0 execution, training, or project completion is claimed yet.

**Comment by TheHalfMoon** [comment]
## MRL-0803 canonical closeout

Fresh post-merge qualification is now complete on the exact canonical merge commit produced by PR #409.

```text
CANONICAL_MAIN_SHA = daa2e83774b7bc210014e4e160bcf1203e943a4e
CANONICAL_MAIN_TREE = dce64b8c75cb10773d21388b7c1cc83be4434b79
MERGE_PARENTS = d584b4e8e80f953b461b0a09ed7dfb00f10888ab, ab435fe970ccd4db7593b6721b8d3c34abd2fdb2
MRL_0803_EVIDENCE_PATH = specs/mesc-research-loop-v1/real-preflight-evidence/MRL-0803.json
MRL_0803_EVIDENCE_SHA256 = 629d8d39a6ea44e753b6d83e07116b12213a74c14f395bea700224fc6781c8e2
MRL_0803 = CLOSED_CANONICAL
```

Post-merge hosted qualification on the original attempts:

```text
CI_RUN = 34709912361 / SUCCESS / exact head daa2e83774b7bc210014e4e160bcf1203e943a4e
quality (py3.12) = SUCCESS
  Pytest = SUCCESS
  MRL machine-state drift/manual-edit gate = SUCCESS
quality (py3.11) = SUCCESS
  Pytest = SUCCESS
  MRL machine-state drift/manual-edit gate = SUCCESS
CODEQL_RUN = 34709912267 / SUCCESS / exact head daa2e83774b7bc210014e4e160bcf1203e943a4e
OPTIONAL_EXTRAS_RUN = 34709912368 / SUCCESS / exact head daa2e83774b7bc210014e4e160bcf1203e943a4e
```

Fresh local exact-main post-merge qualification on the same SHA/tree also completed successfully:

```text
targeted MRL-0803/trust/machine-state tests = 61 passed
MRL machine-state CI gate = 14 passed
docs link hygiene = PASS / 122 Markdown files
medscale check = CLEAN
git diff --check = PASS
worktree = CLEAN
```

One earlier local invocation used the nonexistent script name `scripts/check_markdown_links.py`; that operator invocation error is preserved and was corrected using the canonical workflow command `python scripts/check_docs_links.py`. It is not being hidden or treated as project qualification evidence.

GitHub auto-closed this issue immediately after PR #409 merged; the issue was deliberately reopened because fresh post-merge qualification had not yet completed. That qualification is now genuinely complete, and canonical `origin/main` remains exactly `daa2e83774b7bc210014e4e160bcf1203e943a4e`.

This closeout grants no downstream execution authority beyond completion of MRL-0803. In particular:

```text
MRL-0804 = PLANNED
MODEL_LOADING_AUTHORIZED = FALSE
TOKENIZER_LOADING_AUTHORIZED = FALSE
INFERENCE_AUTHORIZED = FALSE
EXPERIMENT_0_EXECUTION_AUTHORIZED = FALSE
TRAINING_AUTHORIZED = FALSE
TRAINING_READY = FALSE
MRL_REAL_EXPERIMENT_READY = FALSE
PROJECT_COMPLETION = NOT CLAIMED
```

Under Rule R4, MRL-0804 may begin only after this issue is closed `completed`.
